### ISSUE TITLE
azurerm_storage_account error deserializing json

### ISSUE THREAD

**Issue by laurencee** [terraform-providers/terraform-provider-azurerm#3028]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureRM Provider) Version

Terraform v0.11.11
+ provider.azurerm v1.23.0

### Affected Resource(s)

* `azurerm_storage_container`

### Terraform Configuration Files

```hcl
resource "azurerm_resource_group" "default" {
  name     = "myrg"
  location = "westus2"
}

resource "azurerm_storage_account" "default" {
  name                     = "accountname"
  resource_group_name      = "${azurerm_resource_group.default.name}"
  location                 = "westus2"
  account_tier             = "Standard"
  account_kind             = "StorageV2"
  account_replication_type = "LRS"
}

resource "azurerm_storage_container" "default" {
  name                  = "containername"
  resource_group_name   = "${azurerm_resource_group.default.name}"
  storage_account_name  = "${azurerm_storage_account.default.name}"
  container_access_type = "private"
}
```

### Expected Behavior

terraform command succeeds

### Actual Behavior

`azurerm_storage_container.default: Error loading the Resource Groups for Storage Account "accountname": storage.AccountsClient#List: Failure responding to request: StatusCode=200 -- Original Error: Error occurred unmarshalling JSON - Error = 'json: cannot unmarshal string into Go struct field CustomDomain.useSubDomainName of type bool'`

### Steps to Reproduce

1. `terraform apply` or `terraform plan` or `terraform import`

**Comment by darkedges** [comment]
In my case I found that a storage account had 
```
"useSubDomainName": "false" 
```

instead of
```
"useSubDomainName": false
```

Not sure how it was being set.

**Comment by kevinneufeld** [comment]
I am running into the same issue.
```
azurerm_storage_container.default_web_logs: Error loading the Resource Groups for Storage Account "f2bbe251059200cc5f95edd": storage.AccountsClient#List: Failure responding to request: StatusCode=200 -- Original Error: Error occurred unmarshalling JSON - Error = 'json: cannot unmarshal string into Go struct field CustomDomain.useSubDomainName of type bool'
```

**Comment by mikolajzajac** [comment]
getting same issue - rolling back to a previous version did not help either.

**Comment by mikolajzajac** [comment]
found a workaround to this issue, it is related to the AzureRM Provider v2.0: https://github.com/terraform-providers/terraform-provider-azurerm/issues/2807 
tldr: pin the provider version, i.e.
```provider "azurerm" {
  version = "=1.22.0"
}

**Comment by samuel-begin** [comment]
We rolled back to `1.22.1` when we encountered this, but we will need some new features that will be released with `1.24.0`

I'm not sure how we can deal with this yet. Suggestions?

**Comment by mikolajzajac** [comment]
I guess as per https://github.com/terraform-providers/terraform-provider-azurerm/issues/2807: Existing Resources will be required to be imported

**Comment by IvanovOleg** [comment]
This error appears when you have an azure storage account in your subscription which has custom domain feature enabled. It doesn't matter if you manage this account using terraform or not.

**Comment by samuel-begin** [comment]
So i have opened an issue in the `azure-rest-api-specs` repo with [all the details](https://github.com/Azure/azure-rest-api-specs/issues/5626). 
TLDR: The bug in on azure's side. It should be served as a boolean according to their specs

Since I doubt the terraform team wants to account for this, I have made a fork of the provider and will maintain it until the problem is fixed. As of now, the provider is `v1.24.0` with only the type changed for `string`. It is on [this branch](https://github.com/icosamuel/terraform-provider-azurerm/tree/feature/custom-geot-provider) and you can build using `make build-docker` following by replacing the resulting binary in your terraform plugins `~/.terraform.d/plugins/`

Enjoy!

**Comment by tombuildsstuff** [comment]
👋 

Unfortunately this is an issue in the Azure API where it can return a string or a boolean depending on the last time the resource was modified (which is why the Go SDK fails to parse this sometime).

As @samuel-begin has mentioned due to this unfortunately there's not a whole bunch we can do within Terraform (or the Go SDK) to work around this, since the API needs to be fixed to pick one or the other (since updating it to a bool will break if a string is returned, and vica-versa) - as such for the moment I'd suggest +1-ing this issue: https://github.com/Azure/azure-rest-api-specs/issues/5626

Thanks!

**Comment by tombuildsstuff** [comment]
👋 

Looks like the upstream issue's been fixed by no longer returning this field from the API: https://github.com/Azure/azure-sdk-for-go/issues/4473 - as such this should be fixed, so I'm going to close this issue for the moment

Thanks!
